sdk/cosmosdb/cosmos/src/encryption/EncryptionKeyResolver/AzureKeyVaultEncryptionKeyResolver.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import type { TokenCredential } from "@azure/core-auth";
+import type { EncryptionKeyResolver } from "./EncryptionKeyResolver";
+import type { KeyWrapAlgorithm } from "@azure/keyvault-keys";
+import { KeyClient } from "@azure/keyvault-keys";
+
+/**
+ * Implementation of EncryptionKeyResolver that uses Azure Key Vault for customer managed keys.
+ */
+export class AzureKeyVaultEncryptionKeyResolver implements EncryptionKeyResolver {
+  private credentials: TokenCredential;
+
+  constructor(credentials: TokenCredential) {
+    this.credentials = credentials;
+  }
+  /**
+   * wraps the given key using the specified key encryption key path and algorithm.
+   * @param encryptionKeyId - path to the customer managed key to be used for wrapping.
+   * @param algorithm - algorithm to be used for wrapping.
+   * @param key - dek to be wrapped.
+   * @returns wrapped DEK.
+   */
+  public async wrapKey(encryptionKeyId: string, algorithm: string, key: Buffer): Promise<Buffer> {
+    const origin = this.getOrigin(encryptionKeyId);
+    const keyClient = new KeyClient(origin, this.credentials);
```

Comment:
what is the usage pattern of key client? does it needs to be disposed after use?

Reply:
It should be auto disposed off by GC.